Tasks/AzureAppServiceManage/operations/KuduServiceUtils.ts

```diff
@@ -45,17 +45,25 @@ export class KuduServiceUtils {
         outputVariables = outputVariables ? outputVariables : [];
         var outputVariableIterator: number = 0;
         var siteExtensions = await this._appServiceKuduService.getSiteExtensions();
+        var allSiteExtensions = await this._appServiceKuduService.getAllSiteExtensions();
         var anyExtensionInstalled: boolean = false;
         var siteExtensionMap = {};
+        var allSiteExtensionMap = {};
         var extensionLocalPaths: string = "";
         for(var siteExtension of siteExtensions) {
             siteExtensionMap[siteExtension.id] = siteExtension;
         }
+        for(var siteExtension of allSiteExtensions) {
+            allSiteExtensionMap[siteExtension.id] = siteExtension;
+            allSiteExtensionMap[siteExtension.title] = siteExtension;
+        }
 
         for(var extensionID of extensionList) {
             var siteExtensionDetails = null;
-        
-             // Python extensions are moved to Nuget and the extensions IDs are changed. The belo check ensures that old extensions are mapped to new extension ID.
+            if(allSiteExtensionMap[extensionID] && allSiteExtensionMap[extensionID].title == extensionID) {
```

Comment:
(if **not** allSiteExtensionMap[extensionID]) ?

Reply:
we are updating extensionID only when its value is matching title of some siteextension.